src/app/store/reservedip/slice.ts

```diff
@@ -0,0 +1,93 @@
+import type { PayloadAction } from "@reduxjs/toolkit";
+import { createSlice } from "@reduxjs/toolkit";
+
+import type {
+  CreateParams,
+  ReservedIp,
+  ReservedIpState,
+  UpdateParams,
+} from "./types";
+import type { DeleteParams } from "./types/actions";
+import { ReservedIpMeta } from "./types/enum";
+
+import type { GenericItemMeta } from "@/app/store/utils/slice";
+import {
+  generateCommonReducers,
+  generateGetReducers,
+  genericInitialState,
+} from "@/app/store/utils/slice";
+
+const reservedIpSlice = createSlice({
+  name: ReservedIpMeta.MODEL,
+  initialState: genericInitialState as ReservedIpState,
+  reducers: {
+    ...generateCommonReducers<
```

Comment:
I suggest to create common reducers at the top so that they don't have to be recreated multiple times.

`const commonReducers = generateCommonReducers...`

Reply:
Good shout, will do